x-pack/plugins/alerting/server/saved_objects/migrations.test.ts

```diff
@@ -2311,6 +2401,72 @@ describe('successful migrations', () => {
   });
 });
 
+describe('search source migration', () => {
+  it('should apply migration within es query alert rule', () => {
+    const esQueryRuleSavedObject = {
+      attributes: {
+        params: {
+          searchConfiguration: {
+            some: 'prop',
+            migrated: false,
+          },
+        },
+      },
+    } as SavedObjectUnsanitizedDoc;
+
+    const versionToTest = '9.1.3';
+    const migrations = getMigrations(
+      encryptedSavedObjectsSetup,
+      {
+        [versionToTest]: (state) => ({ ...state, migrated: true }),
+      },
+      isPreconfigured
+    );
+
+    expect(
+      migrations[versionToTest](esQueryRuleSavedObject, {} as SavedObjectMigrationContext)
+    ).toEqual({
+      attributes: {
+        params: {
+          searchConfiguration: {
+            some: 'prop',
+            migrated: true,
+          },
+        },
+      },
+    });
+  });
+
+  it('should not apply migration within es query alert rule when searchConfiguration not an object', () => {
+    const esQueryRuleSavedObject = {
+      attributes: {
+        params: {
+          searchConfiguration: 5,
+        },
+      },
+    } as SavedObjectUnsanitizedDoc;
+
+    const versionToTest = '9.1.4';
+    const migrations = getMigrations(
+      encryptedSavedObjectsSetup,
+      {
+        [versionToTest]: (state) => ({ ...state, migrated: true }),
+      },
+      isPreconfigured
+    );
+
+    expect(
+      migrations[versionToTest](esQueryRuleSavedObject, {} as SavedObjectMigrationContext)
+    ).toEqual({
+      attributes: {
+        params: {
+          searchConfiguration: 5,
+        },
+      },
+    });
+  });
+});
+
 describe('handles errors during migrations', () => {
```

Comment:
Since the `getSearchSourceMigrations()` call in the migration is kind of a new "shape" of migration for this SO type, I think it would be nice to have a test for when it throws an error, to make sure we catch the error correctly, etc.

Reply:
Done.